aplib.net-demo/Assets/Scripts/WFC/Cell.cs

```diff
@@ -18,39 +19,59 @@ public class Cell
         /// </summary>
         public List<Tile> Candidates { get; set; }
 
+        /// <summary>
+        /// The x coordinate of this cell, within its grid.
+        /// </summary>
+        public int X { get; }
+
+        /// <summary>
+        /// The y coordinate of this cell, within its grid.
+        /// </summary>
+        public int Y { get; }
+
         /// <summary>
         /// Initializes a new instance of the <see cref="Cell"/> class.
         /// </summary>
-        public Cell()
+        /// <param name="posX">The X coordinate of this cell, within its grid.</param>
+        /// <param name="posY">The Y coordinate of this cell, within its grid.</param>
+        public Cell(int posX, int posY)
         {
+            X = posX;
+            Y = posY;
+
             Tile = new Empty();
             Candidates = new List<Tile>()
             {
                 new Corner(),
```

Comment:
For consistency, add North here as well.

```suggestion
                 new Corner(North),
```

Reply:
Same for the other classes 😀

Reply:
This leaves rider with suggesting you to remove it, but I accept the gray text for consistency sake.